### PR TITLE
Change to Travis CI

### PR DESCRIPTION
**Pull request by jiminhsieh**
I think the problem of the prvious failed builds were no sufficient memory for sbt. At least, this is what I met when I rolled back to CircleCI 2.0.

### CODE REVIEW

**Comment by pathikrit** [comment]
But this build is hanging too above right? It says "Expected — Waiting for status to be reported" (the one that is passing is the v1 build)

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,7 +13,8 @@ jobs:
           keys:
           - build-dependencies-{{ checksum "build.sbt" }}
           - sbt-dependencies--{{ checksum "project/build.properties" }}
-      - run: sbt clean coverage +test
+      # Resource of Docker Executor - 2 vCPUs and 4GB
```
Comment (pathikrit) [inline .circleci/config.yml]: can we add this: `SBT_OPTS: "-XX:+CMSClassUnloadingEnabled -XX:PermSize=256M -XX:MaxPermSize=512M -Xmx2G"`?

**Comment by jiminhsieh** [comment]
> It says "Expected — Waiting for status to be reported"

Could you take a look at this link - https://circleci.com/docs/2.0/workflows-waiting-status/ ? I think it could solve this problem. I did try at my repo with this PR https://github.com/jiminhsieh/better-files/pull/1. I didn't meet the same issue. In the meantime, I didn't enable **Protect this branch**.

File: .circleci/config.yml (hunk 1 above)
Comment (pathikrit) [reply .circleci/config.yml]: That's great! Makes sense!

**Comment by pathikrit** [comment]
But, I do want to protect this branch! If I remove "protect this branch" it removes the option "require status checks"

**Comment by jiminhsieh** [comment]
BTW, we still have 4 months to migrate 2.0. 
> CircleCI 1.0: August 31, 2018 is the Final Day for 1.0 Builds

**Comment by pathikrit** [comment]
screenshot 2018-04-28 12 52 54

This is what I have. I will probably simply switch to Travis CI if this is such a huge issue

**Comment by pathikrit** [comment]
Let's remove Workflows - we don't need it.

**Comment by jiminhsieh** [comment]
I noticed `ci/circleci: build ` still showed `Expected — Waiting for status to be reported`. How about only select `ci/circleci`? It's sufficient to know a PR pass the test or not.

**Comment by pathikrit** [comment]
But `ci/circleci` is the version 1 (since its on master) and `ci/circleci: build` is the version 2 one? If I merge this PR and deselect the latter than no PR will be tested right?

**Comment by jiminhsieh** [comment]
> I am still confused. Look at #232 
> Why is this not building??

I guess the reason of #232 didn't test is @argv-minus-one didn't enable circleci. I can not find any build [here](https://circleci.com/gh/argv-minus-one/better-files). 

At Circle CI, each PR will be **only** tested at **Contributor**'s Circle CI. The concept of dealing PR of Circle CI is different to Travis CI. At Travis CI, PR will be tested at **Owner** side if **Owner** enables it. Contributor's side doesn't have to enable his/her Travis CI. I hope I explain clearly.

**Comment by pathikrit** [comment]
> Simply put, using Circle CI, contributors have to enable Circle CI at their side. Using Travis CI, contributors don't need to care about this.

I see! But, in CircleCI v1 this was not the case right?

**Comment by jiminhsieh** [comment]
> in CircleCI v1 this was not the case right?

I think Circle CI simply doesn't support to build PR at **Owner** side, it's not about which version of Circle CI. You may want to take a look this discussion - [Pull requests not triggering build](https://discuss.circleci.com/t/pull-requests-not-triggering-build/1213). 

I think the best we can do is there are two options:
* Using Circle CI version 2, you have to remind the contributors to enable their Circle CI and turn off `ci/circleci: build`.
* Transfer to Travis CI.

**Comment by pathikrit** [comment]
But I remember CircleCI always used to put a green checkmark if PRs from other authors were submitted. You can see them yourself here: https://github.com/pathikrit/better-files/pulls?q=is%3Apr+is%3Aclosed

**Comment by pathikrit** [comment]
End of day, tell me if this is possible, when someone submits a pull request to better-files, would I see on the PR that the pull request passes `sbt test`???

**Comment by jiminhsieh** [comment]
> I remember CircleCI always used to put a green checkmark if PRs from other authors were submitted.

I believe CircleCI has a huge regression at 2.0. 

> You can see them yourself here: https://github.com/pathikrit/better-files/pulls?q=is%3Apr+is%3Aclosed

I knew this situation.

> tell me if this is possible, when someone submits a pull request to better-files, would I see on the PR that the pull request passes sbt test???

No, it's not possible to see any PR passed `sbt test` at CircleCI for any version if **Contributor** doesn't enabled CircleCI for now.

For example, @argv-minus-one's merged PR #232 and newest PR #238 both used CircleCI **version 1** already. But those two PR are still showing `Expected — Waiting for status to be reported `. As I said before,
> I think Circle CI simply doesn't support to build PR at **Owner side**, it's not about which version of Circle CI.

I think the best we can do is I can transfer CircleCI to Travis CI at this PR. Otherwise you have ask all of **Contributor**s to enable CircleCi.

**Comment by jiminhsieh** [comment]
@pathikrit Could you take a look at CircleCI's **Advanced Settings**? I think the answer to test at **Owner** side would be there. 
screen shot 2018-05-13 at 08 53 52

**Comment by pathikrit** [comment]
Let's just switch to TravisCI. CircleCI seems to be a pain. This file should be easy to migrate to a same `.travis.yml` file:

https://github.com/pathikrit/better-files/blob/master/circle.yml

Is it something you may have time to do?

**Comment by jiminhsieh** [comment]
> Is it something you may have time to do?

Sure, I will finish at the end of this working day.

**Comment by pathikrit** [comment]
Awesome! Thank you!

**Comment by pathikrit** [comment]
@jiminhsieh : I know you are busy, any update on the switch to Travis. This is becoming increasingly painful since I cannot merge PRs confidently. 
e.g. see https://github.com/pathikrit/better-files/pull/239 -- its not building it!

**Comment by jiminhsieh** [comment]
Yeah, I just finished yesterday in my time zone. I will try to explain and leave comments at my code. If there is any problem, I will response after 8 hours(In my time zone, it's already midnight). There are things you need to do yourself. I can not help.

Basically, I followed this  http://eed3si9n.com/auto-publish-from-travis-ci to set up the Travis CI. 

1. `gem install travis`
2. `travis login --auto`
3. Check which version of openssl you use. I used LibreSSL 2.2.7 at Travis CI. You should use the same verion of openssl as Travis CI.(You can change that too) I used this version because of this is what my macOS offered 
3. `travis encrypt-file --repo pathikrit/better-files {your.private.key} {better-files.root.folder}/.travis/travisci-deploy-key.enc`
4. You should get from step 3. somthing similar to this `openssl aes-256-cbc -K $encrypted_1234_key -iv $encrypted_1234_iv -in .travis/deploy_rsa.enc -out ../website_keys/deploy_website_rsa -d`. Please update $encrypted_1234_key and $encrypted_1234_iv to `.travis.yml`.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        wget https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-${LibreSSL_VERSION}.tar.gz -P $HOME
+        tar -xvf $HOME/libressl-${LibreSSL_VERSION}.tar.gz -C $HOME
+        rm $HOME/libressl-${LibreSSL_VERSION}.tar.gz
+        cd $HOME/libressl-${LibreSSL_VERSION}
+        ./configure --prefix=${LibreSSL_HOME} --with-openssldir=${LibreSSL_HOME} && make -j3 && make install
+        cd $OLDPWD
+      fi
+      ${LibreSSL_HOME}/bin/openssl version
+      echo -e "Host github.com\n\tStrictHostKeyChecking no\nIdentityFile ~/.ssh/travisci-deploy-key\n" >> ~/.ssh/config
+      ${LibreSSL_HOME}/bin/openssl aes-256-cbc -K ${fill.your.key} -iv ${fill.your.iv} -in .travis/travisci-deploy-key.enc -out .travis/travisci-deploy-key -d
```
Comment (jiminhsieh) [inline .travis.yml]: You will need to replace `${fill.your.key}` and `${fill.your.iv}`.

File: .travis/travisci-deploy-key.enc [hunk 3]
```diff
@@ -0,0 +1 @@
+{your.encrypted.private.key}
```
Comment (jiminhsieh) [inline .travis/travisci-deploy-key.enc]: ` .travis/travisci-deploy-key.enc` is from this command `travis encrypt-file --repo pathikrit/better-files {your.private.key} {better-files.root.folder}/.travis/travisci-deploy-key.enc`

**Comment by jiminhsieh** [comment]
You also need to add **Environment Variables** at Travis CI.
* CODACY_PROJECT_TOKEN
* SONATYPE_PASSWORD
* SONATYPE_USERNAME
* UPDATEIMPACT_API_KEY

File: .travis.yml (hunk 2 above)
Comment (jiminhsieh) [reply .travis.yml]: Oh, you were right. I missed this part. After I saw your ping, I pushed too soon to clean up. I add this back now.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        wget https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-${LibreSSL_VERSION}.tar.gz -P $HOME
+        tar -xvf $HOME/libressl-${LibreSSL_VERSION}.tar.gz -C $HOME
+        cd $HOME/libressl-${LibreSSL_VERSION}
+        ./configure --prefix=${LibreSSL_HOME} --with-openssldir=${LibreSSL_HOME} && make -j3 && make install
+        cd $OLDPWD
+        rm $HOME/libressl-${LibreSSL_VERSION}.tar.gz && rm -rf $HOME/libressl-${LibreSSL_VERSION}
+      fi
+      ${LibreSSL_HOME}/bin/openssl version
+      echo -e "Host github.com\n\tStrictHostKeyChecking no\nIdentityFile ~/.ssh/travisci-deploy-key\n" >> ~/.ssh/config
+      ${LibreSSL_HOME}/bin/openssl aes-256-cbc -K ${fill.your.key} -iv ${fill.your.iv} -in .travis/travisci-deploy-key.enc -out .travis/travisci-deploy-key -d
```
Comment (jiminhsieh) [inline .travis.yml]: Fail at this line:
> /home/travis/.travis/job_stages: line 70: ${fill.your.key}: bad substitution

https://travis-ci.com/pathikrit/better-files/builds/73775221#L4467

**Comment by pathikrit** [comment]
Why do we need all this SSL stuff? In CircleCI I could just do `sbt publish` and it worked!

**Pull request merged**